src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,473 @@
+            if (identAst.getText().equals(variableDesc.getName())
+                    && !isLeftHandSideValue(identAst)) {
+                variableDesc.registerAsUsed();
+                break;
+            }
+        }
+    }
+
+    /**
+     * Find the scope of variable.
+     *
+     * @param variableDef DetailAst of type {@link TokenTypes#VARIABLE_DEF}
+     * @return scope of variableDef
+     */
+    private static DetailAST findScopeOfVariable(DetailAST variableDef) {
+        final DetailAST result;
+        final DetailAST parentAst = variableDef.getParent();
+        if (parentAst.getType() == TokenTypes.SLIST) {
+            result = parentAst;
+        }
+        else {
+            final DetailAST grandParent = parentAst.getParent();
+            final DetailAST scopeAst = grandParent.findFirstToken(TokenTypes.SLIST);
+            if (scopeAst == null) {
+                result = grandParent;
+            }
+            else {
+                result = scopeAst;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is
+     * used as left-hand side value. An identifier is being used as a left-hand side
+     * value if it is used as the left operand of an assignment or as an
+     * operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return true if identAst is used as a left-hand side value
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return isStandAloneIncrementOrDecrement(identAst)
+                || parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is used as
+     * an operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return true if identAst is used as an operand of stand-alone
+     *         increment or decrement
+     */
+    private static boolean isStandAloneIncrementOrDecrement(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        final DetailAST grandParent = parent.getParent();
+        return TokenUtil.isOfType(parent, INCREMENT_AND_DECREMENT_TOKENS)
+                && TokenUtil.isOfType(grandParent, TokenTypes.EXPR)
+                && isPrefixOrPostfixOnLeftHandSide(identAst);
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is used
+     * in prefix or postfix expression on the left-hand side.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return true if identAst is used in prefix or postfix expression
+     *         on the left-hand side.
+     */
+    private static boolean isPrefixOrPostfixOnLeftHandSide(DetailAST identAst) {
```

Comment:
At this point, we would know that we have an `EXPR` from condition on line 368. It would be good just to pass grandparent from `isStandAloneIncrementOrDecrement` here, and rename parameter to `expression`. This way, we can avoid a bunch of `.getParent()` calls.

Reply:
Resolved.

Reply:
done

Reply:
Resolved.